lib/readyset/query/cached_query.rb

```diff
@@ -0,0 +1,103 @@
+require 'active_model'
+
+module Readyset
+  module Query
+    # Represents a query that is cached by ReadySet.
+    class CachedQuery
+      include ActiveModel::AttributeMethods
+
+      attr_reader :id, :text, :name, :count, :always
+
+      # Returns all of the queries currently cached on ReadySet by invoking the `SHOW CACHES` SQL
+      # extension on ReadySet.
+      #
+      # @return [Array<CachedQuery>]
+      def self.all
+        Readyset.raw_query('SHOW CACHES').map do |result|
+          from_readyset_result(**result.symbolize_keys)
+        end
+      end
+
```

Comment:
Thoughts on extracting this method (and in `ProxyQuery`) to `Query` ? 

Here's an example, but the args should be named more clearly:
```ruby
    def self.find_all_queries(query_command, result_class)
      Readyset.raw_query(query_command).map do |result|
        result_class.send(:from_readyset_result, **result.symbolize_keys)
      end
    end
 ```
    
 Or it could be defined as a module method or something. In `ProxyQuery`, keep the existing method but call `all(args)` instead of it.

Reply:
I agree that this could be simpler, so I actually ended up adding a `Queryable` mixin. Lemme know what you think!